From 595d25ddbc6838eb7ec70a71040246f4c7c5426f Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Tue, 10 Dec 2019 18:08:31 +0100
Subject: Fix RRULE generation for timezones

Summary:
libical 3.0.6 is stricter for RRULEs than previous versions and disallows
setting both the COUNT and UNTIL property at the same time. So don't do
that anymore, one is enough.

Reviewers: dfaure, winterz

Reviewed By: winterz

Subscribers: winterz, kde-pim

Tags: #kde_pim

Differential Revision: https://phabricator.kde.org/D25857
---
 .../Compat-libical3/AppleICal_1.5.ics.ical.ref     | 50 +++++++++++-----------
 .../Evolution_2.8.2_timezone_test.ics.ical.ref     | 12 +++---
 .../data/Compat-libical3/Mozilla_1.0.ics.ical.ref  | 24 +++++------
 autotests/data/Compat/AppleICal_1.5.ics.ical.ref   | 50 +++++++++++-----------
 .../Evolution_2.8.2_timezone_test.ics.ical.ref     |  4 +-
 autotests/data/Compat/Mozilla_1.0.ics.ical.ref     |  8 ++--
 autotests/testicaltimezones.cpp                    |  2 +-
 src/icaltimezones.cpp                              |  1 -
 8 files changed, 75 insertions(+), 76 deletions(-)

diff --git a/autotests/data/Compat-libical3/AppleICal_1.5.ics.ical.ref b/autotests/data/Compat-libical3/AppleICal_1.5.ics.ical.ref
index dca8c16..0173bc8 100644
--- a/autotests/data/Compat-libical3/AppleICal_1.5.ics.ical.ref
+++ b/autotests/data/Compat-libical3/AppleICal_1.5.ics.ical.ref
@@ -17,7 +17,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -68,7 +68,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -119,7 +119,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -170,7 +170,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -221,7 +221,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -272,7 +272,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -323,7 +323,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -374,7 +374,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -425,7 +425,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -476,7 +476,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -527,7 +527,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -578,7 +578,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -629,7 +629,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -680,7 +680,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -731,7 +731,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -782,7 +782,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -833,7 +833,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -884,7 +884,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -935,7 +935,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -986,7 +986,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1037,7 +1037,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1088,7 +1088,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1139,7 +1139,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1190,7 +1190,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1241,7 +1241,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
diff --git a/autotests/data/Compat-libical3/Evolution_2.8.2_timezone_test.ics.ical.ref b/autotests/data/Compat-libical3/Evolution_2.8.2_timezone_test.ics.ical.ref
index 63fdc02..898b2f5 100644
--- a/autotests/data/Compat-libical3/Evolution_2.8.2_timezone_test.ics.ical.ref
+++ b/autotests/data/Compat-libical3/Evolution_2.8.2_timezone_test.ics.ical.ref
@@ -24,7 +24,7 @@ TZNAME:EST
 TZOFFSETFROM:-0400
 TZOFFSETTO:-0500
 DTSTART:19701025T020000
-RRULE:FREQ=YEARLY;UNTIL=20071104T020000;COUNT=37;BYDAY=-1SU;BYMONTH=10
+RRULE:FREQ=YEARLY;UNTIL=20071104T020000;BYDAY=-1SU;BYMONTH=10
 RDATE:20061029T020000
 END:STANDARD
 BEGIN:DAYLIGHT
@@ -32,14 +32,14 @@ TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19760425T020000
-RRULE:FREQ=YEARLY;UNTIL=19870405T020000;COUNT=11;BYDAY=-1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=19870405T020000;BYDAY=-1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19880403T020000
-RRULE:FREQ=YEARLY;UNTIL=20070311T020000;COUNT=19;BYDAY=1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=20070311T020000;BYDAY=1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
@@ -83,7 +83,7 @@ TZNAME:EST
 TZOFFSETFROM:-0400
 TZOFFSETTO:-0500
 DTSTART:19701025T020000
-RRULE:FREQ=YEARLY;UNTIL=20071104T020000;COUNT=37;BYDAY=-1SU;BYMONTH=10
+RRULE:FREQ=YEARLY;UNTIL=20071104T020000;BYDAY=-1SU;BYMONTH=10
 RDATE:20061029T020000
 END:STANDARD
 BEGIN:DAYLIGHT
@@ -91,14 +91,14 @@ TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19760425T020000
-RRULE:FREQ=YEARLY;UNTIL=19870405T020000;COUNT=11;BYDAY=-1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=19870405T020000;BYDAY=-1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19880403T020000
-RRULE:FREQ=YEARLY;UNTIL=20070311T020000;COUNT=19;BYDAY=1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=20070311T020000;BYDAY=1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
diff --git a/autotests/data/Compat-libical3/Mozilla_1.0.ics.ical.ref b/autotests/data/Compat-libical3/Mozilla_1.0.ics.ical.ref
index b443dbb..d9f8078 100644
--- a/autotests/data/Compat-libical3/Mozilla_1.0.ics.ical.ref
+++ b/autotests/data/Compat-libical3/Mozilla_1.0.ics.ical.ref
@@ -23,7 +23,7 @@ TZNAME:EST
 TZOFFSETFROM:-0400
 TZOFFSETTO:-0500
 DTSTART:19701025T020000
-RRULE:FREQ=YEARLY;UNTIL=20071104T020000;COUNT=37;BYDAY=-1SU;BYMONTH=10
+RRULE:FREQ=YEARLY;UNTIL=20071104T020000;BYDAY=-1SU;BYMONTH=10
 RDATE:20061029T020000
 END:STANDARD
 BEGIN:DAYLIGHT
@@ -31,14 +31,14 @@ TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19760425T020000
-RRULE:FREQ=YEARLY;UNTIL=19870405T020000;COUNT=11;BYDAY=-1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=19870405T020000;BYDAY=-1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19880403T020000
-RRULE:FREQ=YEARLY;UNTIL=20070311T020000;COUNT=19;BYDAY=1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=20070311T020000;BYDAY=1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
@@ -82,7 +82,7 @@ TZNAME:EST
 TZOFFSETFROM:-0400
 TZOFFSETTO:-0500
 DTSTART:19701025T020000
-RRULE:FREQ=YEARLY;UNTIL=20071104T020000;COUNT=37;BYDAY=-1SU;BYMONTH=10
+RRULE:FREQ=YEARLY;UNTIL=20071104T020000;BYDAY=-1SU;BYMONTH=10
 RDATE:20061029T020000
 END:STANDARD
 BEGIN:DAYLIGHT
@@ -90,14 +90,14 @@ TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19760425T020000
-RRULE:FREQ=YEARLY;UNTIL=19870405T020000;COUNT=11;BYDAY=-1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=19870405T020000;BYDAY=-1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19880403T020000
-RRULE:FREQ=YEARLY;UNTIL=20070311T020000;COUNT=19;BYDAY=1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=20070311T020000;BYDAY=1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
@@ -141,7 +141,7 @@ TZNAME:EST
 TZOFFSETFROM:-0400
 TZOFFSETTO:-0500
 DTSTART:19701025T020000
-RRULE:FREQ=YEARLY;UNTIL=20071104T020000;COUNT=37;BYDAY=-1SU;BYMONTH=10
+RRULE:FREQ=YEARLY;UNTIL=20071104T020000;BYDAY=-1SU;BYMONTH=10
 RDATE:20061029T020000
 END:STANDARD
 BEGIN:DAYLIGHT
@@ -149,14 +149,14 @@ TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19760425T020000
-RRULE:FREQ=YEARLY;UNTIL=19870405T020000;COUNT=11;BYDAY=-1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=19870405T020000;BYDAY=-1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19880403T020000
-RRULE:FREQ=YEARLY;UNTIL=20070311T020000;COUNT=19;BYDAY=1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=20070311T020000;BYDAY=1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
@@ -200,7 +200,7 @@ TZNAME:EST
 TZOFFSETFROM:-0400
 TZOFFSETTO:-0500
 DTSTART:19701025T020000
-RRULE:FREQ=YEARLY;UNTIL=20071104T020000;COUNT=37;BYDAY=-1SU;BYMONTH=10
+RRULE:FREQ=YEARLY;UNTIL=20071104T020000;BYDAY=-1SU;BYMONTH=10
 RDATE:20061029T020000
 END:STANDARD
 BEGIN:DAYLIGHT
@@ -208,14 +208,14 @@ TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19760425T020000
-RRULE:FREQ=YEARLY;UNTIL=19870405T020000;COUNT=11;BYDAY=-1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=19870405T020000;BYDAY=-1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
 TZOFFSETFROM:-0500
 TZOFFSETTO:-0400
 DTSTART:19880403T020000
-RRULE:FREQ=YEARLY;UNTIL=20070311T020000;COUNT=19;BYDAY=1SU;BYMONTH=4
+RRULE:FREQ=YEARLY;UNTIL=20070311T020000;BYDAY=1SU;BYMONTH=4
 END:DAYLIGHT
 BEGIN:DAYLIGHT
 TZNAME:EDT
diff --git a/autotests/data/Compat/AppleICal_1.5.ics.ical.ref b/autotests/data/Compat/AppleICal_1.5.ics.ical.ref
index 8a48f48..20ae235 100644
--- a/autotests/data/Compat/AppleICal_1.5.ics.ical.ref
+++ b/autotests/data/Compat/AppleICal_1.5.ics.ical.ref
@@ -17,7 +17,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -68,7 +68,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -119,7 +119,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -170,7 +170,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -221,7 +221,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -272,7 +272,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -323,7 +323,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -374,7 +374,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -425,7 +425,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -476,7 +476,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -527,7 +527,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -578,7 +578,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -629,7 +629,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -680,7 +680,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -731,7 +731,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -782,7 +782,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -833,7 +833,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -884,7 +884,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -935,7 +935,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -986,7 +986,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1037,7 +1037,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1088,7 +1088,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1139,7 +1139,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1190,7 +1190,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
@@ -1241,7 +1241,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 END:STANDARD
 BEGIN:STANDARD
 TZNAME:CET
diff --git a/autotests/data/Compat/Evolution_2.8.2_timezone_test.ics.ical.ref b/autotests/data/Compat/Evolution_2.8.2_timezone_test.ics.ical.ref
index 0d238f5..3d9a235 100644
--- a/autotests/data/Compat/Evolution_2.8.2_timezone_test.ics.ical.ref
+++ b/autotests/data/Compat/Evolution_2.8.2_timezone_test.ics.ical.ref
@@ -39,7 +39,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 RDATE;VALUE=DATE-TIME:19950924T030000
 END:STANDARD
 END:VTIMEZONE
@@ -79,7 +79,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 RDATE;VALUE=DATE-TIME:19950924T030000
 END:STANDARD
 END:VTIMEZONE
diff --git a/autotests/data/Compat/Mozilla_1.0.ics.ical.ref b/autotests/data/Compat/Mozilla_1.0.ics.ical.ref
index 086abb6..a194a26 100644
--- a/autotests/data/Compat/Mozilla_1.0.ics.ical.ref
+++ b/autotests/data/Compat/Mozilla_1.0.ics.ical.ref
@@ -38,7 +38,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 RDATE;VALUE=DATE-TIME:19950924T030000
 END:STANDARD
 END:VTIMEZONE
@@ -78,7 +78,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 RDATE;VALUE=DATE-TIME:19950924T030000
 END:STANDARD
 END:VTIMEZONE
@@ -118,7 +118,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 RDATE;VALUE=DATE-TIME:19950924T030000
 END:STANDARD
 END:VTIMEZONE
@@ -158,7 +158,7 @@ TZNAME:CET
 TZOFFSETFROM:+0200
 TZOFFSETTO:+0100
 DTSTART:19790930T030000
-RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9
+RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9
 RDATE;VALUE=DATE-TIME:19950924T030000
 END:STANDARD
 END:VTIMEZONE
diff --git a/autotests/testicaltimezones.cpp b/autotests/testicaltimezones.cpp
index af3a975..71913cb 100644
--- a/autotests/testicaltimezones.cpp
+++ b/autotests/testicaltimezones.cpp
@@ -123,7 +123,7 @@ static const char *VTZ_Prague
       "TZOFFSETFROM:+0200\r\n"
       "TZOFFSETTO:+0100\r\n"
       "DTSTART:19790930T030000\r\n"
-      "RRULE:FREQ=YEARLY;UNTIL=19961027T030000;COUNT=17;BYDAY=-1SU;BYMONTH=9\r\n"
+      "RRULE:FREQ=YEARLY;UNTIL=19961027T030000;BYDAY=-1SU;BYMONTH=9\r\n"
       "RDATE;VALUE=DATE-TIME:19950924T030000\r\n"
       "END:STANDARD\r\n"
       "BEGIN:DAYLIGHT\r\n"
diff --git a/src/icaltimezones.cpp b/src/icaltimezones.cpp
index c6d8891..9c0500f 100644
--- a/src/icaltimezones.cpp
+++ b/src/icaltimezones.cpp
@@ -386,7 +386,6 @@ icalcomponent *ICalTimeZoneParser::icalcomponentFromQTimeZone(const QTimeZone &t
                     icalrecurrencetype r;
                     icalrecurrencetype_clear(&r);
                     r.freq = ICAL_YEARLY_RECURRENCE;
-                    r.count = (year >= 2030) ? 0 : times.count() - 1;
                     r.by_month[0] = month;
                     if (rule & DAY_OF_MONTH) {
                         r.by_month_day[0] = dayOfMonth;
-- 
cgit v1.1
